core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -40,51 +40,31 @@
  * This table may return duplicate rows.
  */
 public class AllDataFilesTable extends BaseMetadataTable {
-  private final TableOperations ops;
```

Comment:
Instead of modifying `metadataLocation` in each metadata table, I added `ops` and `table` to the parent class and implemented needed methods there. I can put the refactoring into a separate PR if that simplifies the review.

Reply:
No longer applies. We need this changes.